src/components/context/TimersContextProvider.tsx

```diff
@@ -0,0 +1,175 @@
+import { stopWorkout } from '../../utils/helpers';
+
+export type Timer = {
+    id: string;
+    status: string;
+    type: string;
+    initialDuration: number;
+    duration: number;
+    initialRounds?: number;
+    rounds?: number;
+    initialRestDuration?: number;
+    restDuration?: number;
+};
+
+export type TimersContextType = {
+    timers: Timer[];
+    running: Timer | Partial<Timer> | null;
+    setRunning: (timer: Timer | Partial<Timer> | null) => void;
+    createTimer: (timer: {
+        status?: string;
+        type: string;
+        duration: number;
+        initialDuration: number;
+        rounds?: number;
+        initialRounds?: number;
+        restDuration?: number;
+        initialRestDuration?: number;
+    }) => Timer;
+    deleteTimer: (id: string) => void;
+    startWorkout: () => void;
+    stopWorkout: () => void;
+    fastForward: () => void;
+};
+
+export const TimersContext = createContext<TimersContextType>({
+    timers: [],
+    running: null,
+    createTimer: () => ({
+        id: '',
+        status: '',
+        type: '',
+        duration: 0,
+        initialDuration: 0,
+        setDuration: () => {},
+    }),
+    setRunning: () => {},
+    deleteTimer: () => {},
+    startWorkout: () => {},
+    stopWorkout: () => {},
+    fastForward: () => {},
+});
+
+const TimersContextProvider = ({ children }: { children: React.ReactNode }) => {
+    const [timers, setTimers] = useState<Timer[]>([]);
+    const [running, setRunning] = useState<Timer | Partial<Timer> | null>(null);
+    const intervalRef: MutableRefObject<number | undefined> = useRef();
+    useEffect(() => {
+        if (running === null) {
+            clearInterval(intervalRef.current);
+        }
+    }, [running]);
+
+    useEffect(() => {
+        return () => {
+            clearInterval(intervalRef.current);
+        };
+    }, []);
+
+    return (
+        <TimersContext.Provider
+            value={{
+                timers,
+                running,
+                setRunning,
+                createTimer: ({ type, duration, initialDuration, rounds, initialRounds, restDuration, initialRestDuration }) => {
+                    const newTimer = {
+                        id: `${Date.now()}`,
+                        status: 'stopped',
+                        type,
+                        duration,
+                        initialDuration,
+                        initialRounds,
+                        rounds,
+                        initialRestDuration,
+                        restDuration,
+                    };
+                    setTimers([...timers, newTimer]);
+                    return newTimer;
+                },
+                deleteTimer: id => setTimers(timers.filter(t => t.id !== id)),
+                startWorkout: () => {
+                    if (running === null) {
+                        intervalRef.current = setInterval(() => {
+                            setTimers(prevTimers => {
```

Comment:
this logic works but is a bit riskier from a developer perspective. There's nothing preventing one from accidentally and incorrectly setting two timers as `running`, which would get the app in a bad state.